test/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(SRCROOT "${PROJECT_SOURCE_DIR}/test/src")
+
+include_directories("${PROJECT_SOURCE_DIR}/include")
+include_directories("${PROJECT_SOURCE_DIR}/extlibs/headers")
+include_directories("${SRCROOT}/TestUtilities")
+
+# System is always built
+SET(SYSTEM_SRC
+    "${SRCROOT}/CatchMain.cpp"
+    "${SRCROOT}/System/Vector2.cpp"
+    "${SRCROOT}/System/Vector3.cpp"
+    "${SRCROOT}/TestUtilities/SystemUtil.hpp"
+    "${SRCROOT}/TestUtilities/SystemUtil.cpp"
+)
+sfml_add_test(systemtest "${SYSTEM_SRC}"  sfml-system)
```

Comment:
I would name the targets differently, this looks different from the conventions used to name modules. For example, sfml-xxx-test (or test-sfml-xxx) would be consistent.

Reply:
Renamed to test-sfml-xyz